### PR TITLE
Fix for Bug 2298

### PR DESCRIPTION
**Pull request by hoyMS**
There is a noway assert in the lower assuming an GT_ARR_ELEM node cannot be a top expression node. This is not true.  We can have a top-level GT_ARR_ELEM. For example, a function call with a parameter of GT_ARR_ELEM can end up being simplified by the inliner to single GT_ARR_ELEM node if the function has an empty body.

The fix here is to remove the assert and use the GT_ARR_ELEM as the inserting point and remove the GT_ARR_ELEM later.

### CODE REVIEW

File: src/jit/lower.cpp [hunk 1]
```diff
@@ -3343,9 +3343,7 @@ Lowering::LowerArrElem(GenTree **ppTree, Compiler::fgWalkData* data)
     GenTreePtr arrObjNode = arrElem->gtArrObj;
     assert(arrObjNode->IsLocal());
 
-    GenTreePtr nextNode = arrElem->gtNext;
-    // ??? Can we have a top-level GT_ARR_ELEM ???
-    noway_assert(nextNode != nullptr);
+    GenTreePtr nextNode = arrElem;
```
Comment (sivarv) [inline src/jit/lower.cpp]: This logic definitely needs some comments for anyone to understand without spending too much of their time.

Comment (hoyMS) [inline src/jit/lower.cpp]: Yes, I'll do that. This is just a temporary test.

Comment (hoyMS) [inline src/jit/lower.cpp]: @sivarv The change is ready to review. Failing tests passed. Thanks.

**Comment by sivarv** [comment]
LGTM

**Pull request merged**